quodlibet/quodlibet/ext/events/synchronizedlyrics.py

```diff
@@ -179,78 +215,184 @@ def _style_lyrics_window(self):
         """.format(self._get_background_color(), self._get_text_color(),
                    self._get_font_size()))
 
+    def _highlight_text(self):
+        return self._get_highlight_color()
+
     def _cur_position(self):
         return app.player.get_position()
 
     def _build_data(self):
+        cur = app.player.info
+        if cur is not None:
+            cur = SongWrapper(cur)
         self.text_buffer.set_text("")
         if app.player.song is not None:
             # check in same location as track
             track_name = app.player.song.get("~filename")
             new_lrc = os.path.splitext(track_name)[0] + ".lrc"
-            print_d("Checking for lyrics file %s" % new_lrc)
+            print_d("Checking for lyrics tag")
             if self._current_lrc != new_lrc:
                 self._lines = []
+                self._words = []
                 if os.path.exists(new_lrc):
-                    print_d("Found lyrics file: %s" % new_lrc)
-                    self._parse_lrc_file(new_lrc)
+                    print_d("Found lyrics file")
+                    self._parse_lrc_file(new_lrc, "file")
+                else:
+                    new_lrc = cur("~lyrics")
+                    print_d("Couldn't find lyrics file, defaulting to tags")
+                    self._parse_lrc_file(new_lrc, "tag")
             self._current_lrc = new_lrc
 
-    def _parse_lrc_file(self, filename):
-        with open(filename, 'r', encoding="utf-8") as f:
-            raw_file = f.read()
+    def _parse_lrc_file(self, lyrics, tagOrFile):
```

Comment:
Very minor thing but `tagOrFile` could be [Enum](https://github.com/quodlibet/quodlibet/blob/30293621f466cbc0076ed2db703702ef0234a853/quodlibet/quodlibet/util/enum.py#L10) called `LyricsSource` or similar. 

Also `_parse_lrc_file` should be renamed since it doesn't just apply to files. Probably better to have a `_parse_lyrics` (takes a lyrics string) called by a `_parse_lyrics_from_source` (which either opens file or tags).